vzmem

```diff
@@ -12,8 +12,8 @@ my $TOP = `top -b -n1 -d0`;
 my $PAGE_SIZE_KB = 4;
 my $MAX_WORD = (`uname -m` =~ /64/? "9223372036854775807" : "2147483647");
 my ($DIR) = grep { -d $_ } ("/etc/sysconfig/vz-scripts", "/etc/vz/conf");
-my ($MEM_KB) = $TOP =~ /^Mem:\s*(\d+)k/mi or die "Cannot parse 'top' command!\n";
-my ($SWP_KB) = $TOP =~ /^Swap:\s*(\d+)k/mi or die "Cannot parse 'top' command!\n";
+my ($MEM_KB) = $TOP =~ /^KiB Mem:\s*(\d+)/mi or die "Cannot parse 'top' command!\n"; 
```

Comment:
Why not. But why would it run better than top?

On Thursday, February 5, 2015, Igor Posledov <notifications@github.com>
wrote:

> In vzmem
> <https://github.com/DmitryKoterov/dklab_vzmem/pull/3#discussion_r24151283>
> :
>
> > @@ -12,8 +12,8 @@ my $TOP = `top -b -n1 -d0`;
> >  my $PAGE_SIZE_KB = 4;
> >  my $MAX_WORD = (`uname -m` =~ /64/? "9223372036854775807" : "2147483647");
> >  my ($DIR) = grep { -d $_ } ("/etc/sysconfig/vz-scripts", "/etc/vz/conf");
> > -my ($MEM_KB) = $TOP =~ /^Mem:\s*(\d+)k/mi or die "Cannot parse 'top' command!\n";
> > -my ($SWP_KB) = $TOP =~ /^Swap:\s*(\d+)k/mi or die "Cannot parse 'top' command!\n";
> > +my ($MEM_KB) = $TOP =~ /^KiB Mem:\s*(\d+)/mi or die "Cannot parse 'top' command!\n";
>
> why not using free command changing code to:
>
> my $TOP = `free`;
> ...my ($MEM_KB) = $TOP =~ /^Mem:\s*(\d+)/mi or die "Cannot parse 'top' command!\n";my ($SWP_KB) = $TOP =~ /^Swap:\s*(\d+)/mi or die "Cannot parse 'top' command!\n";
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DmitryKoterov/dklab_vzmem/pull/3/files#r24151283>.
>